wpilibc/src/generate/main/native/cpp/hid.cpp.jinja

```diff
@@ -89,4 +90,18 @@ bool {{ ConsoleName }}Controller::GetTouchpadPressed() {
 bool {{ ConsoleName }}Controller::GetTouchpadReleased() {
   return GetRawButtonReleased(Button::kTouchpad);
 }
-{% endif %}
+{%- endif %}
+
+void {{ ConsoleName }}Controller::InitSendable(wpi::SendableBuilder& builder) {
+  builder.SetSmartDashboardType("HID");
+  builder.AddStringProperty("ControllerType", [] { return "{{ ConsoleName }}"; }, nullptr);
```

Comment:
@Starlight220 can correct me if I'm wrong, but I'm pretty sure publish const (added in #5158) is applicable here and more efficient (since it doesn't have to repeatedly check the lambda).
```suggestion
  builder.PublishConstString("ControllerType", "{{ ConsoleName }}", nullptr);
```